lua/autorun/owoifier.lua

```diff
@@ -0,0 +1,72 @@
+local OwOemotes = {"OwO", "owo", "UwU", "uwu", "TwT", ">w<", ">wO", "Ow<", ":3c", ":3", ":33", ":333", "nya", "(> owo )>",
+					"(> owo <)", "*pounces on you*", "*glomps u*", "mrowww", "~", ":owo:", ":uwu:", ":awoo:", ":blush:", 
+					":catgirl:", ":pat:", ":snug:", ":thinkerfelix:", ":weebblush1:", ":weebears:", ":widetrap:"} --Add more emotes here
+
+local function OwOifier(text)
+	local splitstring = string.Split(text, "")
+	local isemoteenabled = false
+
+	for k, v in pairs(splitstring) do
+		if v == ":" or v == ";" or v == "[" or v == "]" then --Keep emotes and chat colors the same
+			isemoteenabled = !isemoteenabled
+		elseif v == " " then
+			isemoteenabled = false
+		end
+		if !isemoteenabled then
+			if splitstring[k+1] then
+				local dtext = table.concat(splitstring, "", k, k+1)
+
+				for nyaaa in string.gmatch(dtext, "n[aeiou]+") do --n(vowel) to ny(vowel)
+					local ny = "ny"..string.TrimLeft(nyaaa, "n")
+					local jointext = string.gsub(dtext, nyaaa, ny)
+					table.remove(splitstring, k+1)
+					table.remove(splitstring, k)
+					table.insert(splitstring, k, jointext)
+				end
+			end
+
+			if string.find(v, "[RrLl]") then --R and L to W
```

Comment:
Doing table.remove shifts everything down, then inserting shifts it all back up in O(n) time. Also the Replace parts are overcomplex. You can just:
```
if string.find("RrLl", v) then
if v==v:lower() then
splitstring[k] = "w"
else
splitstring[k] = "W"
end
end
```